src/EditorFeatures/Core/Tagging/AbstractAsynchronousTaggerProvider.TagSource_ProduceTags.cs

```diff
@@ -238,14 +241,18 @@ private void EnqueueWork(bool highPriority, bool frozenPartialSemantics, Cancell
         /// In the event of a cancellation request, this method may <em>either</em> return at the next availability
         /// or throw a cancellation exception.
         /// </remarks>
-        /// <param name="highPriority">
-        /// If this tagging request should be processed as quickly as possible with no extra delays added for it.
+        /// <param name="highPriority"> If this tagging request should be processed as quickly as possible with no extra
+        /// delays added for it.
         /// </param>
-        private async Task<VoidResult> RecomputeTagsAsync(
+        /// <param name="calledFromJtfRun">If this method is being called from within a JTF.Run call.  This is used to
+        /// ensure we don't do unnecessary switches to the threadpool while JTF is waiting on us.</param>
+        private async Task<BufferToTagTree> RecomputeTagsAsync(
             bool highPriority,
             bool frozenPartialSemantics,
+            bool calledFromJtfRun,
```

Comment:
new addition.  want to make sure we dont' do unnecessary hops when in a JTF call.